src/test/java/io/reactivex/internal/operators/flowable/FlowableFromAsyncTest.java

```diff
@@ -0,0 +1,729 @@
+package io.reactivex.internal.operators.flowable;
+
+import org.junit.*;
+import org.reactivestreams.*;
+
+import io.reactivex.*;
+import io.reactivex.exceptions.*;
+import io.reactivex.functions.Consumer;
+import io.reactivex.processors.PublishProcessor;
+import io.reactivex.subscribers.*;
+
+public class FlowableFromAsyncTest {
+
+    PublishAsyncEmitter source;
+
+    PublishAsyncEmitterNoCancel sourceNoCancel;
+
+    TestSubscriber<Integer> ts;
+    
+    @Before
+    public void before() {
+        source = new PublishAsyncEmitter();
+        sourceNoCancel = new PublishAsyncEmitterNoCancel();
+        ts = new TestSubscriber<Integer>(0L);
+    }
+    
+    @Test
+    public void normalBuffered() {
+        Flowable.fromAsync(source, AsyncEmitter.BackpressureMode.BUFFER).subscribe(ts);
+        
+        source.onNext(1);
+        source.onNext(2);
+        source.onComplete();
+        
+        ts.request(1);
+        
+        ts.assertValue(1);
+        
+        Assert.assertEquals(0, source.requested());
+        
+        ts.request(1);
+        
+        ts.assertValues(1, 2);
+        ts.assertNoErrors();
+        ts.assertComplete();
+    }
+
+    @Test
+    public void normalDrop() {
+        Flowable.fromAsync(source, AsyncEmitter.BackpressureMode.DROP).subscribe(ts);
+        
+        source.onNext(1);
+
+        ts.request(1);
+
+        source.onNext(2);
+        source.onComplete();
+        
+        ts.assertValues(2);
```

Comment:
It's [assertValues(T... values)](https://github.com/ReactiveX/RxJava/blob/1.x/src/main/java/rx/observers/TestSubscriber.java#L618).